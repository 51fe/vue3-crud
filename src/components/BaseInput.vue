<script setup lang="ts">

withDefaults(
  defineProps<{
    modelValue?: string
    placeholder?: string
  }>(),
  {
    modelValue: undefined,
    placeholder: '请输入'
  }
)

const emit = defineEmits<{
  (e: 'update:modelValue', modelValue: string): void
}>()
</script>
<template>
  <el-input
    :model-value="modelValue"
    :max-length="200"
    :placeholder="placeholder"
    clearable
    class="base-input"
    @input="emit('update:modelValue', $event)"
  />
</template>
<style>
.base-input {
  width: 100%;
}
</style>
