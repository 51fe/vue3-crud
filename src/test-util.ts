import { render, RenderOptions, RenderResult } from '@testing-library/vue'

import ElementPlus from 'element-plus'
import CN from 'element-plus/lib/locale/lang/zh-cn'

export const customRender = (
  ui: any, overrides?: Omit<RenderOptions, 'global'>): RenderResult => {
  return render(ui, {
    global: {
      plugins: [[ElementPlus, ...[{ locale:  CN }]]]
    },
    ...overrides
  })
}
export { defineComponent, ref, reactive, nextTick } from 'vue'
export * from '@testing-library/vue'
export { customRender as render }