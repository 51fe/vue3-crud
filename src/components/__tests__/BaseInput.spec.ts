import { fireEvent, screen, render } from '../../test-util'

import BaseInput from '../BaseInput.vue'

test('renders correctly by default', () => {
  const { html, getByPlaceholderText } = render(BaseInput)
  getByPlaceholderText('请输入')
  expect(html()).toMatchSnapshot()
})

test('changes props correctly', () => {
  const modelValue = 'hello'
  const placeholder = '请输入姓名'
  render(BaseInput, {
    props: {
      modelValue,
      placeholder
    }
  })
  screen.getByPlaceholderText(placeholder)
  screen.getByDisplayValue(modelValue)
})

test('emits input event correctly', async () => {
  const { emitted } = render(BaseInput)
  const value = 'world'
  const input = screen.getByRole('textbox')
  await fireEvent.update(input, value)
  expect(emitted()['update:modelValue']).toEqual([[value]])
})