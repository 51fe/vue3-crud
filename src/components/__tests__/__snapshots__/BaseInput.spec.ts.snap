// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly by default 1`] = `
"<div class="el-input el-input--suffix base-input">
  <!-- input -->
  <!-- prepend slot -->
  <!--v-if-->
  <div class="el-input__wrapper">
    <!-- prefix slot -->
    <!--v-if--><input class="el-input__inner" max-length="200" type="text" autocomplete="off" tabindex="0" placeholder="请输入"><!-- suffix slot -->
    <!--v-if-->
  </div><!-- append slot -->
  <!--v-if-->
</div>"
`;
